dali/pipeline/operators/geometric/bb_flip.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/pipeline/operators/geometric/bb_flip.h"
+
+namespace dali {
+
+DALI_REGISTER_OPERATOR(BbFlip, BbFlip, CPU);
+
+
+DALI_SCHEMA(BbFlip)
+                .DocStr(R"code(Operator for horizontal flip (mirror) of bounding box.
+                               Input: Bounding box coordinates; in either (x1,y1,w,h)
+                               or (x1,y1,x2,y2) format)code")
+                .NumInput(1)
+                .NumOutput(1)
+                .AddArg("coordinates_type",
+                        R"code(True, for width-height representation.
+                               False for two-point representation.)code",
+                        DALI_BOOL);
+
+
+BbFlip::BbFlip(const dali::OpSpec &spec) :
+        Operator<CPUBackend>(spec),
+        coordinates_type_wh_(spec.GetArgument<bool>("coordinates_type")) {
+}
+
+
+void BbFlip::RunImpl(dali::SampleWorkspace *ws, const int idx) {
+  auto &input = ws->Input<CPUBackend>(idx);
+  auto input_data = input.data<float>();
+
+  DALI_ENFORCE(input.size() == BB_TYPE_SIZE, "Bounding box in wrong format");
+  DALI_ENFORCE(input.type().id() == DALI_FLOAT || input.type().id() == DALI_FLOAT16,
+               "Bounding box in wrong format");
+
+  DALI_ENFORCE([](const float *data, size_t size) -> bool {
+      for (size_t i = 0; i < size; i++) {
+        if (data[i] < 0 || data[i] > 1.0)
+          return false;
+      }
+      return true;
+  }(input_data, input.size()), "Not all bounding box parameters are in [0.0, 1.0]");
```

Comment:
BBoxes are currently being passed as [x,y,w,h]. Will this account for it? As far as I see you expect these values to be (0.0-1.0)

Reply:
Yes, width and height are also in 0.0-1.0 range. Is it a problem?